src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -115,4 +122,49 @@ public void validate(@Nullable Schema inputSchema) {
       }
     }
   }
+
+  private void validatePartitionProperties(@Nullable Schema schema) {
+    Table table = BigQueryUtil.getBigQueryTable(getProject(), getDataset(), getTable(), getServiceAccountFilePath());
+    if (table != null) {
+      StandardTableDefinition tableDefinition = table.getDefinition();
+      TimePartitioning timePartitioning = tableDefinition.getTimePartitioning();
+      if (timePartitioning == null && createPartitionedTable != null && createPartitionedTable) {
+        throw new InvalidConfigPropertyException(String.format("Destination table '%s' does not support partitioning",
+                                                               table.getTableId().getTable()),
+                                                 "Create Partitioned Table");
+      }
+      if (timePartitioning != null && timePartitioning.getField() != null
+        && !timePartitioning.getField().equals(partitionByField)) {
+        throw new InvalidConfigPropertyException(String.format("Destination table '%s' is partitioned by column '%s'",
```

Comment:
Destination table '%s' is partitioned by column '%s'. Please set the partition field to '%s'.

Reply:
Done.